### PR TITLE
Feat/dashboard

### PR DESCRIPTION
**Pull request by DaniilVolobuiev**
## Describe your changes

> feat: Dashboard. Doctor can see today appointment in dashboard. They are grouped by start and end time. Initially doctor receives 10 appointments, by clicking load more doctor can load all today appointments.

## Issue ticket code (and/or) and link

> [Link to JIRA ticket](https://smartdc23.atlassian.net/jira/software/projects/SMAR/boards/1/backlog?selectedIssue=SMAR-68)
screenshot-1684331558040

### CODE REVIEW

File: src/components/Dashboard/AppointmentElement/index.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            <Stack direction="row" gap="5px">
+              <Typography
+                fontSize={VERY_SMALL_FONT_SIZE}
+                fontWeight="700"
+                color={ANOTHER_FUCKING_BLUE}
+              >
+                {patientName}
+              </Typography>
+              <Typography fontSize={VERY_SMALL_FONT_SIZE} fontWeight="500">
+                {getPatientInfo()}
```
Comment (lyudad) [inline src/components/Dashboard/AppointmentElement/index.tsx]: you will recalculate it with each rerender, change

File: src/components/Dashboard/AppointmentElement/index.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            </Typography>
+            {fullText.length > 100 && !show ? (
+              <Typography display="inline">{fullText}</Typography>
+            ) : (
+              <p>{cuttedText}</p>
+            )}
+          </Box>
+          {fullText.length > 100 && (
+            <Box color={ACTIVE} onClick={() => setShow(!show)}>
+              {t('Show')} {show ? 'less' : 'more'}
```
Comment (lyudad) [inline src/components/Dashboard/AppointmentElement/index.tsx]: why we have text in component?

File: src/components/Scheduler/index.tsx [hunk 3]
```diff
@@ -12,7 +13,10 @@ import TimezoneSelect from './TimezoneSelect/TimezoneSelect';
 import { WHITE } from '@constants/colors';
 import { useAppDispatch, useAppSelector } from '@redux/hooks';
 import { ToastContainer, toast } from 'react-toastify';
-import { Availability, availabilityApi } from '../../services/AvailabilityService';
+import {
+  Availability,
+  availabilityApi,
+} from '../../services/AvailabilityService';
```
Comment (lyudad) [inline src/components/Scheduler/index.tsx]: absolute path

File: src/components/Dashboard/AppointmentElement/index.tsx (hunk 1 above)
Comment (DaniilVolobuiev) [reply src/components/Dashboard/AppointmentElement/index.tsx]: now all calculations are using useMemo

File: src/components/Dashboard/AppointmentElement/index.tsx (hunk 2 above)
Comment (DaniilVolobuiev) [reply src/components/Dashboard/AppointmentElement/index.tsx]: fixed

File: src/components/Scheduler/index.tsx (hunk 3 above)
Comment (DaniilVolobuiev) [reply src/components/Scheduler/index.tsx]: not my code, fixed

File: src/components/Dashboard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      {}
+    );
+  }, [appointmentsArray, sortedAppointments]);
+  if (!appointmentsArray) {
+    return (
+      <div>
+        <Skeleton />
+      </div>
+    );
+  }
```
Comment (lyudad) [inline src/components/Dashboard/index.tsx:47]: I realy affraid what are you doing here, or very smart or very complicated.  move att least to separate hook

Comment (DaniilVolobuiev) [reply src/components/Dashboard/index.tsx:47]: moved logic to hook

File: src/components/Scheduler/index.tsx [hunk 5]
```diff
@@ -30,14 +31,15 @@ export interface IScheduleItem {
 
 const calendarStyle = {
   backgroundColor: WHITE,
-  height: '600px'
+  height: '600px',
 };
```
Comment (lyudad) [inline src/components/Scheduler/index.tsx:35]: why styles in component?

**Pull request merged**